recipe-client-addon/lib/NormandyDriver.jsm

```diff
@@ -156,6 +156,59 @@ this.NormandyDriver = function(sandboxManager) {
       sandboxManager.removeHold(`setTimeout-${token}`);
     },
 
+    /*
+     * Return a promise that resolves to an Addon ID if installation is successful.
+     */
+    installAddon(url) {
+      function getInstallForURLPromise(installUrl) {
+        return new Promise(function(resolve, reject) {
+          AddonManager.getInstallForURL(installUrl, resolve, "application/x-xpinstall");
+        });
+      }
+
+      // Return a promise
+      return getInstallForURLPromise(url).then((installObject) => {
+        return new Promise(function(resolve, reject) {
+          ((innerInstallObject, innerResolve, innerReject) => {
+            innerInstallObject.install();
+            innerInstallObject.addListener({
+              onInstallEnded(addonInstall, addon) {
+                const addonId = addon.id;
+                innerResolve(addonId);
+              },
+              onInstallFailed(addonInstall) {
+                innerReject(`AddonInstall error code: [${addonInstall.error}]`);
+              },
+            });
+          })(installObject, resolve, reject);
+        });
+      });
+    },
+
+    /*
+     * Return a promise that resolves to a success messsage if
+     * addon uninstall is successful.
+     */
+    uninstallAddon(addonId) {
```

Comment:
Making this an async function and using the promise returned by getAddonByID would make this a lot cleaner.